juju/juju
# fix: do not fail probes during controller outage

This fixes a long standing issue where a connection to the juju controller is lost
and a k8s unit fails readiness and liveness probes, even though the workload is
still healthy and running.

## QA steps

- Bootstrap k8s
- Add a model
- Deploy postgresql-k8s with trust and 3 units
- Wait for healthy postgresql via juju status
- Scale the controller stateful set to 0 pods
- Wait a few minutes
- Check `kubectl get pods` on namespace where postgresql is deployed.
- All pods should be ready with 0 restarts.

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2036594

**Jira card:** JJUJU-7257
